cache/guava-cache/src/main/java/org/jdbi/v3/cache/guava/GuavaCache.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.cache.guava;
+
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.CacheStats;
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.UncheckedExecutionException;
+import org.jdbi.v3.core.cache.JdbiCache;
+import org.jdbi.v3.core.cache.JdbiCacheBuilder;
+import org.jdbi.v3.core.cache.JdbiCacheLoader;
+
+/**
+ * Cache implementation using the guava {@link Cache} object.
```

Comment:
The Guava authors caution you against using this class. From the `CacheBuilder` javadoc:

**Prefer Caffeine over Guava's caching API**

In particular, there are known deadlock bugs in this Cache implementation.

Is there a specific use case for including a Guava implementation of JdbiCache? I think we should skip it if not - it is effectively unmaintained.

Reply:
I did not know that. Interesting that they themselves point at caffeine. I am open to keep this out; we don't use any fancy features of guava, so I am not sure that we would hit those bugs. Having this plugin means you can reduce your dependency footprint by just using guava, not guava + caffeine.

Reply:
I don't think you need to use particularly fancy features to get hurt. Take a look at the bug tracker:
https://github.com/google/guava/issues?q=is%3Aopen+is%3Aissue+label%3Apackage%3Dcache+label%3Atype%3Ddefect

Simple things like calling `invalidateAll()` concurrently with a `LoadingCache` load can deadlock the cache forever.

Reply:
This was the reason I pulled caffeine in in the first place - to avoid all the known problems with the Guava cache.

Reply:
I believe the deadlocks and other problems were introduced when Java 8 computations were retrofitted on top. That did not work well with the existing logic and is still not production safe imho. There were minor bugs or quirks that were mostly benign, such as `invalidateAll()` not being linearizable. Caffeine's tests run against Guava's Cache for a compatibility check, which is mostly a proxy except for computes due to its problems. I think their cache is fine for Java 5ish legacy code, but as unmaintained and they do not accept bug fixes it shouldn't be preferred going forward.